examples/misc/demo_ribbon_box.py

```diff
@@ -57,45 +59,35 @@ def draw(self, renderer, *args, **kwargs):
         super().draw(renderer, *args, **kwargs)
 
 
-if True:
-    from matplotlib.transforms import Bbox, TransformedBbox
-    from matplotlib.ticker import ScalarFormatter
-
-    # Fixing random state for reproducibility
-    np.random.seed(19680801)
-
+def main():
     fig, ax = plt.subplots()
 
     years = np.arange(2004, 2009)
-    box_colors = [(0.8, 0.2, 0.2),
-                  (0.2, 0.8, 0.2),
-                  (0.2, 0.2, 0.8),
-                  (0.7, 0.5, 0.8),
-                  (0.3, 0.8, 0.7),
-                  ]
-    heights = np.random.random(years.shape) * 7000 + 3000
-
-    fmt = ScalarFormatter(useOffset=False)
-    ax.xaxis.set_major_formatter(fmt)
+    heights = [7900, 8100, 7900, 6900, 9800]
+    box_colors = [
+        (0.8, 0.2, 0.2),
+        (0.2, 0.8, 0.2),
+        (0.2, 0.2, 0.8),
+        (0.7, 0.5, 0.8),
+        (0.3, 0.8, 0.7),
+    ]
 
     for year, h, bc in zip(years, heights, box_colors):
         bbox0 = Bbox.from_extents(year - 0.4, 0., year + 0.4, h)
         bbox = TransformedBbox(bbox0, ax.transData)
-        rb_patch = RibbonBoxImage(bbox, bc, interpolation="bicubic")
+        ax.add_artist(RibbonBoxImage(ax, bbox, bc, interpolation="bicubic"))
+        ax.annotate(str(h), (year, h), va="bottom", ha="center")
 
-        ax.add_artist(rb_patch)
-
-        ax.annotate(r"%d" % (int(h/100.)*100),
-                    (year, h), va="bottom", ha="center")
+    ax.set_xlim(years[0] - 0.5, years[-1] + 0.5)
+    ax.set_ylim(0, 10000)
 
-    patch_gradient = BboxImage(ax.bbox, interpolation="bicubic", zorder=0.1)
     gradient = np.zeros((2, 2, 4))
```

Comment:
Add a comment `# background gradient`.

Even though the example could use a bit more explanation, I'm not enforcing it as part of this PR, because this is an incremental improvement. However, adding this small comment at least helps to understand what this code segment is about.

Reply:
I just renamed the variable.